packages/interaction/src/InteractionManager.js

```diff
@@ -945,6 +953,20 @@ export default class InteractionManager extends EventEmitter
         }
     }
 
+    /**
+     * Puts a event on a queue to be dispatched later. This is used to guarantee correct
+     * ordering of over/out events.
+     *
+     * @param {PIXI.Container|PIXI.Sprite|PIXI.TilingSprite} displayObject - the display object in question
+     * @param {string} eventString - the name of the event (e.g, mousedown)
+     * @param {object} eventData - the event data object
+     * @private
+     */
+    delayDispatchEvent(displayObject, eventString, eventData)
+    {
+        this.delayedEvents.push([displayObject, eventString, eventData]);
```

Comment:
Reduce the amount of objects/arrays that are created by pushing to a 1-dimensional list.
```suggestion
        this.delayedEvents.push(displayObject, eventString, eventData);
```

Reply:
What you're suggesting would be slightly faster in V8 (which I believe can deal with that), but significantly slower in SpiderMonkey, which later likely won't be able to tell your array element's shapes as easily anymore (you're pushing stuff of at best 3 different shapes there).

The overall fastest way in all engines (if you really cared about perf) would be to just use an object, instead of confusing them with arrays containing varying stuff.

For the 3-4 elements we're pushing to that array, pretty much everything would be premature optimization though.

I played around a bit to confirm my suspicions, and it turns out that chrome is "pretty fast" in all cases, but Firefox kind of loses it if you try to put everything in 1 array:

![5523](https://user-images.githubusercontent.com/1431530/63163525-ec29dc00-c025-11e9-9cd2-14972913efaa.png)

https://jsperf.com/test-array-vs-array-in-array

Reply:
Then prefer object over array seems like a fair compromise.